src/clj/harja/palvelin/palvelut/tyomaapaivakirja.clj

```diff
@@ -65,13 +71,80 @@
   (oikeudet/vaadi-lukuoikeus oikeudet/raportit-tyomaapaivakirja user (:urakka-id tiedot))
   (hae-kommentit db tiedot))
 
-(defn- tallenna-kommentti [db user tiedot]
+(defn- tallenna-kommentti [db user tiedot fim api-sahkoposti kehitysmoodi?]
   (oikeudet/vaadi-kirjoitusoikeus oikeudet/raportit-kommentit user (:urakka-id tiedot))
-  (tyomaapaivakirja-kyselyt/lisaa-kommentti<! db {:urakka_id (:urakka-id tiedot)
-                                                  :tyomaapaivakirja_id (:tyomaapaivakirja_id tiedot)
-                                                  :versio (:versio tiedot)
-                                                  :kommentti (:kommentti tiedot)
-                                                  :luoja (:id user)})
+  (let [{tyomaapaivakirjan-paivamaara :paivamaara
+         urakka-nimi :urakka-nimi
+         hallintayksikko-id :hallintayksikko-id
+         urakka-id :urakka-id
+         tyomaapaivakirja_id :tyomaapaivakirja_id
+         versio :versio
+         kommentti :kommentti} tiedot
+        vastaus (tyomaapaivakirja-kyselyt/lisaa-kommentti<! db {:urakka_id urakka-id
+                                                                :tyomaapaivakirja_id tyomaapaivakirja_id
+                                                                :versio versio
+                                                                :kommentti kommentti
+                                                                :luoja (:id user)})
+
+        tyomaapaivakirjan-paivamaara (pvm/palvelimen-aika->suomen-aikaan tyomaapaivakirjan-paivamaara)
+        tyomaapaivakirjan-paivamaara (pvm/fmt-p-k-v-lyhyt tyomaapaivakirjan-paivamaara)
+
+        onnistui? (some? (:kommentti vastaus))
+        sampo-id (urakka-kyselyt/hae-urakan-sampo-id db (:urakka-id tiedot))
+        roolit #{"urakan vastuuhenkilö"}
+        vastaanottajat (when fim (fim/hae-urakan-kayttajat-jotka-roolissa fim sampo-id roolit))
+        urakanvalvoja? (roolit/roolissa? user roolit/urakanvalvoja)]
+
+    ;; Jos kommentoija on urakanvalvoja, lähetetään sähköposti ilmoitus kaikille urakan vastuuhenkilöille
+    ;; Vastaanottajat:  
+    ;; ({:kayttajatunnus, :sahkoposti, :puhelin, :sukunimi :roolit [Urakan vastuuhenkilö], :roolinimet [vastuuhenkilo], :poistettu, :etunimi, :tunniste, :organisaatio}
+    ;; { ... }
+    ;; { ... } )
+    (if
+      ;; Tarkista onnistuiko tallennus, onko lähettäjä urakanvalvoja, ollaanko tuotannossa ja onko vastaanottajia olemassa
+      (and
+        sampo-id
+        onnistui?
+        urakanvalvoja?
+        (not kehitysmoodi?)
+        (some? vastaanottajat))
+      (try
+        (do
+          (doseq [henkilo vastaanottajat]
+            (let [{sahkoposti :sahkoposti} henkilo
+                  viestin-otsikko "Työmaapäiväkirjassa uusi kommentti"
+                  harja-url "https://extranet.vayla.fi/harja/" ;; http://localhost:3000/
```

Comment:
Voisikohan urlin passata frontista suoraan niin ei tarvitse kovakoodata, onko meillä olemassa mitään @nav/polku ?

Reply:
Ei ole olemassa mitään, mutta tällaisen saa tehtyä näin: `(str js/window.location.protocol "//" js/window.location.host js/window.location.pathname)`

Reply:
Kommentin tallennus tehdään kyllä frontin kautta josta lähetämme parametrit clj puolelle, joten voimme kyllä passata tuon harjan window locationin. Laitoin tuollaisen virvelin: 

https://github.com/finnishtransportagency/harja/pull/3207/commits/be0f8af6281f9c60a3d146c0728834db16f3cdbf